include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -70,125 +95,301 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
         int m_nknots;
 
     public:
+        /// @brief The type of the discrete dimension representing the B-splines.
         using discrete_dimension_type = NonUniformBSplines;
 
+        /// @brief The type of a DiscreteDomain whose elements identify the B-splines.
         using discrete_domain_type = DiscreteDomain<DDim>;
 
+        /// @brief The type of a DiscreteElement identifying a B-spline.
         using discrete_element_type = DiscreteElement<DDim>;
 
+        /// @brief The type of a DiscreteVector representing an "index displacement" between two B-splines.
         using discrete_vector_type = DiscreteVector<DDim>;
 
         Impl() = default;
 
-        template <class OriginMemorySpace>
-        explicit Impl(Impl<DDim, OriginMemorySpace> const& impl)
-            : m_domain(impl.m_domain)
-            , m_nknots(impl.m_nknots)
-        {
-        }
-
-        /// @brief Construct a `Impl` using a brace-list, i.e. `Impl bsplines({0., 1.})`
+        /** @brief Constructs an Impl using a brace-list, i.e. `Impl bsplines({0., 1.})`
+         *
+         * The brace-list is the list of break points. It is used to build the DiscreteDomain indexing
+         * the knots and iterated over to build the knots coordinates list and initialize the associated DiscreteSpace.
```

Comment:
Needs updating. Ideally if you know how I think the easiest is to reference the constructor which does the work

Reply:
I added the @see but I am not sure if it fulfills your asking.

Reply:
Yes this looks good. It correctly generates a link to the more complete docstring. This still needs updating though as it contains unnecessary information. I would recommend something like:
```suggestion
         * Constructs a Impl by iterating over a list of break points. Internally this constructor calls the constructor
         * ... . More details of the implementation can be found there.
```
Or just repeat the information from the other docstring here